app/grandchallenge/challenges/models.py

```diff
@@ -582,7 +582,7 @@ def assign_permissions(self):
         self.assign_forum_permissions()
 
     def assign_forum_permissions(self):
-        if self.display_forum_link and not self.hidden:
+        if self.display_forum_link:
```

Comment:
This was approved in the previous PR, but I'm actually wondering if it makes sense now. Shouldn't hidden challenges be able to have a forum for their users as well? 

I misunderstood the current permission assignment for forums. Currently, hidden challenges are allowed to have forums, and both participants and admins can access those forums. 

Assuming that we want to keep it that way, this removes the self.hidden check.

Reply:
Yes, hidden challenges should have forums but they should not be visible to non-participants. I guess the same goes for public challenges?

Reply:
Yes ,at the moment only admins and participants of a challenge can view the challenge's forum, Non-participants and anonymous users cannot. I think that makes sense. 

At the moment, I believe that public challenges also have a public forum, but I actually think that is a bit weird. Locking it down to participants makes more sense to me.